### ISSUE TITLE
CCSD(T) doesn't work when reading an FCIDUMP file?

### ISSUE THREAD

**Issue by ndattani** [pyscf/pyscf#1577]
CCSD and CCSD(T) work when doing an RHF calculation from scratch, but if I load an FCIDUMP file instead:
```
mf = pyscf.tools.fcidump.to_scf(name+'.fcidump', molpro_orbsym=False, mf=None)
mf.run()
```
Then CCSD works but the (T) correction does not:
```
cycle = 7  E_corr(CCSD) = -0.0493362837053574  dE = -8.8611082e-08  norm(t1,t2) = 2.4739e-06
    CPU time for CCSD iter      3.82 sec, wall time      0.07 sec
    CPU time for CCSD     27.04 sec, wall time      0.49 sec
CCSD converged
E(CCSD) = -75.01232875126405  E_corr = -0.04933628370535739
    CPU time for CCSD integral transformation      3.84 sec, wall time      0.07 sec
Overwritten attributes  get_ovlp get_hcore  of <class 'pyscf.scf.hf_symm.SymAdaptedRHF'>
/home/nike/.local/lib/python3.8/site-packages/pyscf/gto/mole.py:1193: UserWarning: Function mol.dumps drops attribute energy_nuc because it is not JSON-serializable
  warnings.warn(msg)
Traceback (most recent call last):
  File "./inp_withDump.py", line 39, in <module>
    et = mcc.ccsd_t() # works only when not reading FCIDUMP file.
  File "/home/nike/.local/lib/python3.8/site-packages/pyscf/cc/ccsd.py", line 1129, in ccsd_t
    return ccsd_t.kernel(self, eris, t1, t2, self.verbose)
  File "/home/nike/.local/lib/python3.8/site-packages/pyscf/cc/ccsd_t.py", line 51, in kernel
    orbsym = _sort_eri(mycc, eris, nocc, nvir, eris_vvop, log)
  File "/home/nike/.local/lib/python3.8/site-packages/pyscf/cc/ccsd_t.py", line 140, in _sort_eri
    orbsym = symm.addons.label_orb_symm(mol, mol.irrep_id, mol.symm_orb,
  File "/home/nike/.local/lib/python3.8/site-packages/pyscf/symm/addons.py", line 70, in label_orb_symm
    s_mo = numpy.dot(s, mo)
  File "<__array_function__ internals>", line 180, in dot
ValueError: shapes (0,0) and (7,6) not aligned: 0 (dim 1) != 7 (dim 0)
```
I gave full input and output files at MMSE: [Why does CCSD work, but not CCSD(T), after reading an FCIDUMP in PySCF?](https://mattermodeling.stackexchange.com/questions/10277/why-does-ccsd-work-but-not-ccsdt-after-reading-an-fcidump-in-pyscf).

A folder containing the input and output files can also be found at [Modeling Matters](https://github.com/HPQC-LABS/Modeling_Matters/tree/master/10277) or [QC Bugs](https://github.com/HPQC-LABS/QCBugs/issues/20).

**Comment by xwang862** [comment]
FCIDUMP file doesn't contain explicit molecule or basis set information, thus the attempt in [this line](https://github.com/pyscf/pyscf/blob/c0c4dbcb97b5725b5bbec3257b78f18c63663c76/pyscf/symm/addons.py#L69) to rebuild overlap matrix (`s` in `s_mo = numpy.dot(s, mo)`) from molecule returned an empty array, leading to this error. 

This should be fixed in PySCF I believe, but a quick workaround is to assign your prebuilt molecule `mol` to your `mf.mol`:
```
import pyscf
from pyscf import gto, scf, ao2mo, cc, tools

mol = gto.Mole()
mol.atom = ''' O
               H  1  0.9576257
               H  1  0.9576257  2 104.51 '''
mol.unit = 'angstrom'
mol.basis = 'STO-3G'
mol.charge = 0
mol.spin = 0
mol.verbose = 9
mol.symmetry = True
mol.symmetry_subgroup = 'C2v'
name = 'out'
#mol.output = name+'.txt' # only worked up to end of SCF (CC output went to stdout)
mol.build()

mf = tools.fcidump.to_scf(name+'.fcidump', molpro_orbsym=False, mf=None)
mf.mol = mol  # <== Assign prebuilt 'mol' to the new mf 
mf.run()
```

**Comment by ndattani** [comment]
Thanks @xwang862! I'll leave this open since it's a bug that should be fixed in PySCF, even if it's as simple as adding the logic: 
```
#Psudocode:
IF using fcidump THEN mf.mol = mol
```
CCSD(T) should be possible to calculate with nothing but an FCIDUMP (it doesn't need to know the name of the molecule or basis set) .

**Comment by xwang862** [comment]
Turns out my previous workaround `mf.mol = mol` was a bad idea - it worked for STO-3G, but failed for cc-pVDZ and TZ (unconverged SCF). The reason is that the re-assigned `mf.mol` is inconsistent with some attributes, such as `get_ovlp`, in the `mf` that is recovered from FCIDUMP (read [code](https://github.com/pyscf/pyscf/blob/c0c4dbcb97b5725b5bbec3257b78f18c63663c76/pyscf/tools/fcidump.py#L310) for more details). Sorry I didn't test enough...

I suggested a simple solution in pullreq #1580.
